src/ProjectFileIO.cpp

```diff
@@ -1723,17 +1799,19 @@ bool ProjectFileIO::WriteDoc(const char *table,
                              const char *schema /* = "main" */)
 {
    auto db = DB();
+
+   TransactionScope transaction(GetConnection(), "UpdateProject");
```

Comment:
You do this to avoid an autocommit for each chunk?  You discovered the need for it or you knew?  This page isn't explicit about it https://www.sqlite.org/c3ref/blob_write.html but autocommit is mentioned for blob close.

You might comment.

Reply:
I want to make sure that the project is written atomically. It was guaranteed previously that the project is saved atomically, so I kept this behavior.